aws-lc-rs/src/rsa/encryption.rs

```diff
@@ -0,0 +1,510 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0 OR ISC
+
+use crate::{
+    encoding::{AsDer, Pkcs8V1Der, PublicKeyX509Der},
+    error::{KeyRejected, Unspecified},
+    fips::indicator_check,
+    ptr::{DetachableLcPtr, LcPtr},
+};
+use aws_lc::{
+    EVP_PKEY_CTX_new, EVP_PKEY_CTX_set0_rsa_oaep_label, EVP_PKEY_CTX_set_rsa_mgf1_md,
+    EVP_PKEY_CTX_set_rsa_oaep_md, EVP_PKEY_CTX_set_rsa_padding, EVP_PKEY_decrypt,
+    EVP_PKEY_decrypt_init, EVP_PKEY_encrypt, EVP_PKEY_encrypt_init, EVP_sha1, EVP_sha256,
+    EVP_sha384, EVP_sha512, OPENSSL_malloc, EVP_MD, EVP_PKEY, EVP_PKEY_CTX, RSA_PKCS1_OAEP_PADDING,
+};
+use core::{fmt::Debug, mem::size_of_val, ptr::null_mut};
+use mirai_annotations::verify_unreachable;
+
+use super::{
+    encoding,
+    key::{generate_rsa_key, is_rsa_key, key_size_bits, key_size_bytes},
+    KeySize,
+};
+
+/// RSA-OAEP with SHA1 Hash and SHA1 MGF1
```

Comment:
Question: can customers define there own mixed modes? Or are these the only ones supported by aws-lc-rs? For the record I like starting with only theses since matching hashes is the best practice though other libraries support mixing so we might need them in the future for compatibility.

Reply:
No, they can only use the ones provided by aws-lc-rs. The `OaepAlgorithm` type has private struct fields with no constructor.